src/tetris/field/field.ts

```diff
@@ -79,33 +94,78 @@ export default class Field {
 	private readonly _width: integer;
 	private readonly _height: integer;
 	private readonly _drawOffset: Vector2;
+	private readonly _scoreText: Phaser.GameObjects.Text;
 	private readonly _brickFactory: BrickFactory;
 
 	private _nextActiveBrickDrop: number;
 	private _activeBrickDropInterval: number = 400;
+	private _score: number = 0;
+
+	private _fieldState: FieldState = FieldState.Playing;
 
 	// contains only stuck bricks
 	private readonly _bricks: Brick[];
-	private readonly _state: boolean[][];
+	private readonly _blocks: Block[][];
 	//endregion
 
 	//region private methods
 	private _generateNewBrick(time: number): void {
 		this._activeBrick = this._brickFactory.newBrick(this);
 		this._nextActiveBrickDrop = time + this._activeBrickDropInterval;
+		if (this._activeBrick.checkIfStuck()) {
+			this._fieldState = FieldState.Loss;
+		}
 	}
 
 	private _setupField(): void {
-		const iterator = this._state.keys();
+		const iterator = this._blocks.keys();
 		for (let key of iterator) {
-			this._state[key] = new Array(this._width).fill(false);
+			this._blocks[key] = new Array(this._width).fill(null);
 		}
 	}
 
 	private _addToField(blocks: Block[], brickOffset: Vector2): void {
 		blocks.forEach(block => {
-			this._state[brickOffset.x + block.position.x][brickOffset.y + block.position.y] = true;
+			this._blocks[block.currentPosition.x][block.currentPosition.y] = block;
 		});
 	}
+
+	private _deleteCompletedRows(): number {
+		let rowsDeleted = 0;
+		for (let y = 0; y < this.height; y++) {
```

Comment:
All these for loops do not really look beautiful.
Consider to use `for...in` or `forEach`.
Also, nesting level is to high in this function (>4).
One suggestion would be to extract methods out of the body of this function in order to reduce function length and nesting level.

Reply:
I refactored this function, which took quite some effort (see commits).